src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2StreamWindowManager.cs

```diff
@@ -204,6 +209,14 @@ internal void OnDataOrHeadersReceived(Http2Connection connection)
                 {
                     if (initial) _initialBurst--;
 
+                    // When sendWindowUpdateBeforePing is true, try to send a WINDOW_UPDATE to make Google backends happy.
+                    // Unless we are doing the initial burst, do not send PING if we were not able to send the WINDOW_UPDATE.
+                    // See point 3. in the comments above the class definition for more info.
+                    if (sendWindowUpdateBeforePing && !connection.ForceSendConnectionWindowUpdate() && !initial)
```

Comment:
nit: There is an if check on initial immediately above. I'd put this if into an else and remove the need to test initial again.

Reply:
With that the code would not attempt to `connection.ForceSendConnectionWindowUpdate()` in case `initial == true`.

Reply:
With that the code would not attempt to `connection.ForceSendConnectionWindowUpdate()` in case `initial == true`, which is not strictly necessary, but IMO still a desirable behavior.

Reply:
I mean like this:
```cs
if (initial)
{
	_initialBurst--;
}
else
{
	// When sendWindowUpdateBeforePing is true, try to send a WINDOW_UPDATE to make Google backends happy.
	// Unless we are doing the initial burst, do not send PING if we were not able to send the WINDOW_UPDATE.
	// See point 3. in the comments above the class definition for more info.
	if (sendWindowUpdateBeforePing && !connection.ForceSendConnectionWindowUpdate())
    {
        return;
    }
}
```
That's how I'd write it, but they achieve the same thing.

Reply:
It does. Since `&&` evaluation order is left to right, `connection.ForceSendConnectionWindowUpdate()` will be invoked, and potentially send an WUD, even if `initial==true`. The code you are suggesting is not equivalent to the current one.

Am I missing something or misunderstanding you?